pkg/ingester/ingester.go

```diff
@@ -263,6 +282,19 @@ type Ingester struct {
 	tenantsWithOutOfOrderEnabledStat   *expvar.Int
 	minOutOfOrderTimeWindowSecondsStat *expvar.Int
 	maxOutOfOrderTimeWindowSecondsStat *expvar.Int
+
+	// Read path memory threshold in bytes
+	readPathMemoryThreshold uint64
+	// Read path CPU threshold as a fraction of 1
+	readPathCPUThreshold float64
```

Comment:
Unfortunately, I don't think we can separate read vs write CPU utilization without changes to the go runtime, or thread pinning. We'd have to have a single threshold and measurement for the entire process.